Facebook Accounts

bjoshua18 opened the pull request:
Implementamos la vinculación de cuentas de facebook
Creamos los tests de los modelos

bjoshua18 committed:
Facebook Accounts
Implementamos la vinculación de cuentas de facebook
Creamos los tests de los modelos

alejandrogq commented on app/controllers/facebook_accounts_controller.rb:
```diff
@@ -0,0 +1,18 @@
+
+  def destroy
+    @facebook_account = Current.user.facebook_accounts.find_by(id: params[:id])
+    @facebook_account.destroy
```
Y si esto devuelve false?

alejandrogq commented on app/controllers/posts_controller.rb:
```diff
@@ -1,47 +1,47 @@
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @posts = Current.user.posts.order_by(publish_at: 1)
```
Has creado un indice para el campo publish_at

bjoshua18 replied:
Es el index donde se listan todos los posts por orden de fecha de publicación

alejandrogq replied:
No me has entendido, un indice en la base de datos

bjoshua18 replied:
Ok, ¿un índice para publish_at para que la consulta order_by sea más rápida?

```
# post.rb
index({publish_at: -1})
```

alejandrogq commented on app/controllers/posts_controller.rb:
```diff
@@ -1,47 +1,47 @@
+  end
+
+  def destroy
+    @post.destroy
```
Si falla el destroy?

alejandrogq commented on app/models/post.rb:
```diff
@@ -28,6 +28,7 @@ def published?
 
   def publish_to_twitter!
     tweet = twitter_account.client.update(body)
+    # tweet = twitter_account.client.update_with_media(body, File.new(Rails.root.join('app', 'assets', 'images', 'page_flow.png')))
```
Esto se quedo comentado

alejandrogq commented on app/models/user.rb:
```diff
@@ -8,6 +8,7 @@ class User
 
   has_many :twitter_accounts
+  has_many :facebook_accounts
   has_many :posts
```
Post es solo de facebook?

bjoshua18 replied:
El modelo Post tiene que ser refactorizado. Tenemos pendiente cambiar la estructura de los accounts, que TwitterAccount y FacebookAccount hereden de una superclase Account, así, Post pertenece a un account mas genérico

alejandrogq replied:
ok

alejandrogq commented on app/views/facebook_accounts/index.html.haml:
```diff
@@ -0,0 +1,13 @@
+  .d-flex.align-items-center.mb-4
+    .me-4
+      = image_tag acc.image, class: 'rounded-circle me-4'
+      -# = link_to acc.name, acc.account.get_object('me'), class: 'link-primary', target: '_blank'
```
comentado

alejandrogq commented on app/views/sessions/new.html.haml:
```diff
@@ -1,11 +1,11 @@
 %h1 Login
 
 = form_with url: login_path do |form|
```
Usar simple_form?

bjoshua18 committed:
Refactorización
Modificaciones de métodos para prevenir errores
Eliminación de código comentado
Cambio del login form con simple_form

bjoshua18 committed:
Fix
Arreglado el bug del session_controller con los parámetros del simple_form

bjoshua18 committed:
Creación de índice para post con publish_at
